include/flamegpu/gpu/CubTemporaryMemory.cuh

```diff
@@ -0,0 +1,24 @@
+#ifndef INCLUDE_FLAMEGPU_GPU_CUBTEMPORARYMEMORY_CUH_
```

Comment:
`CubTemporaryMemory` should be in `detail`, it is not part of the user-interface (this is also true of things like CUDAScanCompaction, but that isn't being added by this PR). 

Alternatively the header could be placed in the `src` directory, but that would depend on where it is included from (so would probably require more substantial changes)

Reply:
Most of the `gpu` sub dir is the same. Have addressed this.